internal/operator-controller/controllers/common_controller.go

```diff
@@ -27,6 +27,31 @@ import (
 	ocv1 "github.com/operator-framework/operator-controller/api/v1"
 )
 
+const (
+	// maxConditionMessageLength is the Kubernetes API limit (32768) minus truncation suffix length (25 characters)
+	// This ensures final message with suffix never exceeds the API limit: 32768 - 25 = 32743
+	maxConditionMessageLength = 32743
```

Comment:
If we already account for the suffix here, we will be truncating messages that are still less than the 32768 length.
I.e. we should only truncate if the message is > than the whole size limit (rather than size limit - suffix length).

On line 44 we already take the truncation suffix into account when generating the truncated message and ensuring its at most the max size.